crates/oxc_linter/src/rules/jest/prefer_todo.rs

```diff
@@ -0,0 +1,270 @@
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jest(prefer-todo): Suggest using `test.todo`.")]
+#[diagnostic(severity(warning))]
+struct UmImplementedTestDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferTodo;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// When test cases are empty then it is better to mark them as `test.todo` as it
+    /// will be highlighted in the summary output.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// This rule triggers a warning if empty test cases are used without 'test.todo'.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// test('i need to write this test'); // invalid
+    /// test('i need to write this test', () => {}); // invalid
+    /// test.skip('i need to write this test', () => {}); // invalid
+    ///
+    /// test.todo('i need to write this test');
+    /// ```
+    PreferTodo,
+    restriction,
+);
+
+impl Rule for PreferTodo {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::CallExpression(call_expr) = node.kind() {
+            let counts = call_expr.arguments.len();
+
+            if counts < 1
+                || should_filter_case(call_expr)
+                || !is_string_type(&call_expr.arguments[0])
+                || !is_type_of_jest_fn_call(
+                    call_expr,
+                    node,
+                    ctx,
+                    &[JestFnKind::General(JestGeneralFnKind::Test)],
+                )
+            {
+                return;
+            }
+
+            if counts == 1 && !filter_todo_case(call_expr) {
+                let (content, span) = get_fix_content(call_expr);
+                ctx.diagnostic_with_fix(UmImplementedTestDiagnostic(span), || {
+                    Fix::new(content, span)
+                });
+            }
+
+            if is_empty_function(call_expr) {
+                let (_, span) = get_fix_content(call_expr);
+                ctx.diagnostic_with_fix(EmptyTest(span), || {
+                    let (content, span) = build_code(call_expr, ctx);
+                    Fix::new(content, span)
+                });
+            }
+        }
+    }
+}
+
+fn filter_todo_case(expr: &CallExpression) -> bool {
+    if let Expression::MemberExpression(mem_expr) = &expr.callee {
+        if let Some(name) = mem_expr.static_property_name() {
+            return name == "todo";
+        }
+    }
+    false
+}
+
+fn should_filter_case(expr: &CallExpression) -> bool {
+    let result = match &expr.callee {
+        Expression::Identifier(ident) => ident.name.starts_with('x') || ident.name.starts_with('f'),
+        _ => false,
+    };
+    result || filter_todo_case(expr)
+}
+
+fn is_string_type(arg: &Argument) -> bool {
+    matches!(
+        arg,
+        Argument::Expression(Expression::StringLiteral(_) | Expression::TemplateLiteral(_))
+    )
+}
+
+fn is_empty_function(expr: &CallExpression) -> bool {
```

Comment:
Is this hurt performance? Maybe we can try only check the second argument:

```js
test('foo', () => {})
```

Reply:
Let me try.